### ISSUE TITLE
Can sideberry respect tab hidings from other extensions?

### ISSUE THREAD

**Issue by TeNNoX** [mbnuqw/sidebery#191]
## Description
I have installed [Simple Tab Groups](https://github.com/drive4ik/simple-tab-groups), but when I open a group it would hide all other tabs in the tab bar, but not in Sideberry.

I'd still love to have the tree structure of Sideberry, I just felt the group management of Sideberry not easy enough to quickly switch /navigate between groups somehow.

If Sideberry would respect the "hiding requests" of other extensions, I think it could be a great compromise :man_shrugging:

**Comment by mbnuqw** [comment]
Yeah, I think this is possible, although I need to do more tests.

![Peek 2020-04-06 19-46](https://user-images.githubusercontent.com/6276694/78571273-70dc9b00-783f-11ea-97e5-bf001c62bff7.gif)

In sidebery the closest analogy for the group of Simple Tab Groups is the panel. You can set name, icon, bind container, etc... just in case :3

**Comment by TeNNoX** [comment]
I appreciate your quick response. :+1: 
I realised I can't to without the tree structure (I'm just opening too many tabs :sweat_smile: ), so I will give sideberry panels another try.

I just had some annoyances with the panels - but I will give them a bit more attention an create an issue if I find something.

**Comment by TeNNoX** [comment]
Wow... I took the time to reset & go through all settings - and now I have nothing left to wish for.
This extension is A-MA-ZING! :heart: 

Do you accept donations?

**Comment by mbnuqw** [comment]
Thank you.

I'll check later if this feature not affect on the performance I'll add it, since it may be useful for someone.

**Comment by gee-forr** [comment]
Hey there,

I would also love it if SideBerry would only show tabs of the current container I'm in. I use [Conex to only show tabs from my current container](https://github.com/kesselborn/conex) and [TabCenter Redux which unfortunately has not seen an update in 2 years](https://github.com/eoger/tabcenter-redux). I'd love to use something more comprehensive and feature packed like SideBerry, but I have so many tabs across so many containers, that the only way to be effective is to show only the current containers tabs at any time, which Conex+TCR does well.

Hope you're able to find a way to make SideBerry support this. Thanks for this fantastic addon.

**Comment by novirium** [comment]
This feature would still be quite useful to me - I'm in a similar situation, using Simple Tab Groups. I use it primarily to hide whole groups/workspaces of tabs at once, and then open them in various windows - something Sidebery panels can't replicate at the moment. The functionality shown in the comment above is exactly what I'd need to use both extensions together:

> Yeah, I think this is possible, although I need to do more tests.
> 
> ![Peek 2020-04-06 19-46](https://user-images.githubusercontent.com/6276694/78571273-70dc9b00-783f-11ea-97e5-bf001c62bff7.gif)

If I was to try and use Sidebery panels for this instead, the main extra feature needed would be the mutual exclusion and tab consistency between windows: Other than the "default" panel, opening a panel in one window when it was already open in another would just refocus the other window. Switching away from a panel in one window (hiding it) and then switching _to_ it in another window would shift all of its tabs over to the new window. I get this is probably out of scope for this extension though - just wanted to help provide some context for why people might want this.

**Comment by acicovic** [comment]
Hi there,

I just started trying Sidebery.

I'm using Simple Tab Groups along with Tree Style Tab, with about 15 groups and 295 tabs.

I will try Sidebery panels, but a setting to make Sidebery obey to Simple Tab Groups groups would be great to get started, especially for users with many tabs.

**Comment by acicovic** [comment]
A follow-up:

Unless I have missed something, there seems to be no way to display panels in a type of dropdown like Simple Tab Groups does. In my scenario this makes panels hard to use.

**Comment by drod3763** [comment]
I have the same issue using Simple Tab Groups. Right now I have too many tabs open to switch groups over to Sidebery but if I get them under control I may do so. Submitted an upvote. Hopefully we can get some traction on this one.

**Comment by brightghost** [comment]
> Yeah, I think this is possible, although I need to do more tests.
> 
> ![Peek 2020-04-06 19-46](https://user-images.githubusercontent.com/6276694/78571273-70dc9b00-783f-11ea-97e5-bf001c62bff7.gif)
> 
> In sidebery the closest analogy for the group of Simple Tab Groups is the panel. You can set name, icon, bind container, etc... just in case :3

I'm confused about the status of this. Is the behavior shown in this image possible with the version of the extension currently available? For me it is showing all the tabs from every group together and I haven't managed to make it behave with any of the relevant settings.

**Comment by MrQwerky** [comment]
For me, tab grouping is an absolutely essential part of my workflow.  As such, I have been using Simple Tab Groups for a very long time, and am now trying out Sidebery.

I have the Sidebery navigation bar set to vertical, so there is lots of room for the panel icons.  I meticulously created a Sidebery panel to match each of my STG groups, and then dragged the entire STG group of tabs to that Sidebery panel.  I find that when I click on a panel/tab in Sidebery, the FF native tab bar automatically changes to the correct STG group and selects the correct tab in that group.  Likewise, selecting a group/tab in STG causes Sidebery to automatically select the corresponding panel/tab.  This is exactly the behaviour I desire, so I hope that if any changes are made, they will be made optional.

So far, it seems that Sidebery and STG can co-exist nicely, except for the fact the every time FF is started, STG throws up a tab saying that there is a conflicting add-on, requiring me to close that tab.  I now have CSS in place to hide the native FF tab bar when Sidebery is active, and to display the native FF tab bar when Sidebery is not active.  Incidently, I have the sidebar on the right side of the FF window, with the navigation bar on the right side of the sidebar, and the container-colour bar on the left side of the tab, next to the favicon.  For me, this provides a pleasant UI.

**Comment by critbase** [comment]
This would be great as an option for me; I currently use [Tiled Tab Groups](https://addons.mozilla.org/en-US/firefox/addon/tiled-tab-groups/), but I find it doesn't integrate that well with Sidebery because it accomplishes "groups" by hiding tabs that are not in the group you switch to.

**Comment by tvvoty** [comment]
I also use STG, so if this feature gets to be added, that would be great.

**Comment by AkiVonAkira** [comment]
Any update on this? Would love to use both Simple Tab Groups and Sideberry together

**Comment by Michaelknubben** [comment]
Same, I'm interested if this is still a possibility!

**Comment by tech189** [comment]
As a workaround, if instead you want to import your STG tab groups into Sidebery, I wrote a Python program to convert a Simple Tab Groups backup to a Sidebery snapshot.

Instructions and code here: https://gist.github.com/tech189/252d9e9ec8942ada0a52c041e24a0e12
Relevant discussion here: https://github.com/mbnuqw/sidebery/discussions/1456

**Comment by bossanovaorca** [comment]
this is the only missing feature that's holding me back from migrating from Tree Style Tabs
